crates/evm/evm/src/executors/invariant/mod.rs

```diff
@@ -440,27 +612,32 @@ impl<'a> InvariantExecutor<'a> {
                         call_result,
                         &state_changeset,
                     )
-                    .map_err(|e| TestCaseError::fail(e.to_string()))?;
+                    .map_err(|e| eyre!(e.to_string()))?;
                     if !result.can_continue || current_run.depth == self.config.depth - 1 {
                         invariant_test.set_last_run_inputs(&current_run.inputs);
                     }
                     // If test cannot continue then stop current run and exit test suite.
                     if !result.can_continue {
-                        return Err(TestCaseError::fail("Test cannot continue."))
+                        break 'stop;
                     }
 
                     invariant_test.set_last_call_results(result.call_result);
                     current_run.depth += 1;
                 }
 
                 // Generates the next call from the run using the recently updated
-                // dictionary.
-                current_run.inputs.push(
-                    invariant_strategy
-                        .new_tree(&mut invariant_test.execution_data.borrow_mut().branch_runner)
-                        .map_err(|_| TestCaseError::Fail("Could not generate case".into()))?
-                        .current(),
-                );
+                // dictionary if initial seq isn't `depth` long.
+                if current_run.depth as usize > initial_seq.len().saturating_sub(1) {
```

Comment:
we should apply this logic for discarded inputs as well, otherwise the corpus will get stuck at the same discarded sequence and won't advance (`invariant::test_invariant_assume_does_not_revert` test failure)
```rust
if discarded || (current_run.depth as usize > initial_seq.len().saturating_sub(1)) {
```